kubedriver/kegd/manager.py

```diff
@@ -219,12 +219,30 @@ def __expand_helm_action(self, deploy_task, kegd_files, render_context):
             namespace = self.templating.render(deploy_action.namespace, render_context)
         deploy_action.namespace = namespace
         if deploy_action.values != None:
-            deploy_action.values = self.templating.render(deploy_action.values, render_context)
-            values_file_path = kegd_files.get_helm_file(deploy_action.values)
-            with open(values_file_path, 'r') as f:
-                values_file_content = f.read()
-            rendered_values_file_content = self.__process_template(values_file_content, render_context, values_file_path)
-            deploy_action.values = rendered_values_file_content
+            expanded_valuefiles = []
+            for filepath in deploy_action.values:
+                rendered_path = self.templating.render(filepath, render_context)
+                kegd_path = kegd_files.get_helm_file(rendered_path)
+                values_file_content = None
+                with open(kegd_path, 'r') as f:
+                    values_file_content = f.read()
+                rendered_values_file_content = self.__process_template(values_file_content, render_context, kegd_path)
+                expanded_valuefiles.append(rendered_values_file_content)
+            deploy_action.values = expanded_valuefiles
+        if deploy_action.setfiles != None:
+            expanded_setfiles = {}
+            for key in deploy_action.setfiles:
+                if key in expanded_setfiles:
+                    raise InvalidDeploymentStrategyError(f'Duplicate key:{key} present in setfiles')
+                filepath = deploy_action.setfiles[key]
+                rendered_path = self.templating.render(filepath, render_context)
+                kegd_path = kegd_files.get_helm_file(rendered_path)
+                file_content = None
+                with open(kegd_path, 'r') as f:
+                    file_content = f.read()
+                # Custom files likely won't have templates, render content here later if needed
+                expanded_setfiles[key] = file_content
```

Comment:
Let's process this as a template now. I can see what you mean about it being less likely but it's a one liner which would provide a lot of flexibility, if someone needed to template a large config file etc.

Reply:
Sure